packages/foam-vscode/src/features/tags-tree-view/index.ts

```diff
@@ -29,7 +30,7 @@ export class TagsProvider implements vscode.TreeDataProvider<TagTreeItem> {
 
   private tags: {
     tag: string;
-    noteUris: vscode.Uri[];
+    notes: Note[];
```

Comment:
@riccardoferretti Ok that sounds like a good argument. By reading the file content you mean directly something like `fs.read()`, right? Or is there a foam way to access the content? Though that would require probably passing the `foam` instance to the tree children which might not be ideal.

I'm kind of hesitant to use `fs.read` directly as it feels like an expensive operation to perform over an unknown and possibly large number of files. However there is a way for the tooltip to resolve to a promise instead of having to be pre computed, so we would only read the file content on hover. That'd mean having a min version of 1.52.1 though.

Reply:
the right way to do that would be via the `Datastore`, but yeah, that's probably not being passed in all the way.

I am a big fan of dependency injection, but I gotta admit that at times it gets in the way 😅 
Open to other approaches (after all that's also what vscode does with its packages), although I wouldn't make that change part of this PR as a whole

Reply:
I refactored by passing the data store to the tree provider, see latest commit